File: apps/mobibix-web/src/services/operations.api.ts
import { authenticatedFetch, extractData } from "./auth.api";

export type DailyClosingStatus = "DRAFT" | "SUBMITTED" | "REOPENED";
export type CashVarianceStatus = "PENDING" | "APPROVED" | "REJECTED";
export type StockVerificationStatus = "DRAFT" | "CONFIRMED" | "CANCELLED";
export type ShiftClosingStatus = "OPEN" | "CLOSED";
export type CashClosingMode = "DAILY_ONLY" | "SHIFT_AND_DAILY";
export type AdjustmentReason =
  | "BREAKAGE"
  | "DAMAGE"
  | "LOST"
  | "INTERNAL_USE"
  | "CORRECTION"
  | "SPARE_DAMAGE";

export interface ExpenseCategory {
  id: string;
  name: string;
  isDefault: boolean;
  shopId?: string;
}

export interface CashLeakageAnalysis {
  difference: number;
  severity: "NONE" | "LOW" | "MEDIUM" | "HIGH";
  suggestions: string[];
}

// ─── DAILY CLOSING ────────────────────────────────────────────────────────────

export interface DailySummary {
  date: string;
  shopId: string;
  status: DailyClosingStatus | "OPEN";
  openingCash: number;
  salesCash: number;
  salesUpi: number;
  salesCard: number;
  salesBank: number;
  cashWithdrawFromBank: number;
  cashDepositToBank: number;
  otherCashIn: number;
  otherCashOut: number;
  totalIn: number;
  totalOut: number;
  supplierPaymentsCash: number;
  expenseCash: number;
  expectedClosingCash: number;
  reportedClosingCash: number;
  cashDifference: number;
  shiftInfo?: {
    totalShifts: number;
    closedShifts: number;
    shifts: ShiftClosing[];
  } | null;
}

export interface ShiftClosing {
  id: string;
  date: string;
  shiftName: string;
  openingCash: number;
  reportedClosingCash?: number;
  cashDifference?: number;
  status: ShiftClosingStatus;
  openedBy: string;
  closedBy?: string;
  closedAt?: string;
}

export interface DailyClosing {
  id: string;
  shopId: string;
  date: string;
  status: DailyClosingStatus;
  openingCash: number;
  salesCash: number;
  salesUpi: number;
  salesCard: number;
  salesBank: number;
  expectedClosingCash: number;
  reportedClosingCash: number;
  cashDifference: number;
  varianceReason?: string;
  varianceNote?: string;
  closedBy?: string;
  closedAt?: string;
  createdAt: string;
}

export interface CashVariance {
  id: string;
  shopId: string;
  dailyClosingId: string;
  expectedCash: number;
  physicalCash: number;
  difference: number;
  reason: string;
  reportedBy: string;
  approvedBy?: string;
  approvedAt?: string;
  status: CashVarianceStatus;
  notes?: string;
  createdAt: string;
  dailyClosing?: { date: string };
}

export async function getDailySummary(
  shopId: string,
  date: string
): Promise<DailySummary> {
  const res = await authenticatedFetch(
    `/mobileshop/reports/daily-summary?shopId=${shopId}&date=${date}`
  );
  if (!res.ok) throw new Error("Failed to fetch daily summary");
  return extractData(res);
}

export async function getCashLeakageAnalysis(
  shopId: string,
  date: string
): Promise<CashLeakageAnalysis> {
  const res = await authenticatedFetch(
    `/mobileshop/reports/cash-leakage-analysis?shopId=${shopId}&date=${date}`
  );
  if (!res.ok) throw new Error("Failed to fetch leakage analysis");
  return extractData(res);
}

export async function closeDay(payload: {
  shopId: string;
  date: string;
  mode: "SYSTEM" | "MANUAL";
  reportedClosingCash: number;
  manualEntries?: {
    salesCash?: number;
    salesUpi?: number;
    salesCard?: number;
    salesBank?: number;
    otherCashIn?: number;
    cashWithdrawFromBank?: number;
    expenseCash?: number;
    supplierPaymentsCash?: number;
    otherCashOut?: number;
    cashDepositToBank?: number;
  };
  denominations?: Record<string, number>;
  varianceReason?: string;
  varianceNote?: string;
}): Promise<DailyClosing> {
  const res = await authenticatedFetch(`/cash/daily-close`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload),
  });
  if (!res.ok) {
    const err = await extractData(res);
    throw new Error((err as any).message || "Failed to close day");
  }
  return extractData(res);
}

export async function reopenDay(payload: {
  shopId: string;
  date: string;
  reason: string;
}): Promise<DailyClosing> {
  // Note: Backend might need a specific reopen endpoint in CashController if not reusing legacy
  const res = await authenticatedFetch(`/operations/daily-closing/reopen`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload),
  });
  if (!res.ok) {
    const err = await extractData(res);
    throw new Error((err as any).message || "Failed to reopen day");
  }
  return extractData(res);
}

export async function getDailyClosings(
  shopId: string
): Promise<DailyClosing[]> {
  const res = await authenticatedFetch(`/cash/daily-history?shopId=${shopId}`);
  if (!res.ok) throw new Error("Failed to fetch closings");
  return extractData(res);
}

// ─── SHIFT MANAGEMENT ──────────────────────────────────────────────────────────

export async function openShift(payload: {
  shopId: string;
  shiftName: string;
}): Promise<ShiftClosing> {
  const res = await authenticatedFetch(`/cash/shift/open`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload),
  });
  if (!res.ok) {
    const err = await extractData(res);
    throw new Error((err as any).message || "Failed to open shift");
  }
  return extractData(res);
}

export async function closeShift(payload: {
  shopId: string;
  reportedClosingCash: number;
}): Promise<ShiftClosing> {
  const res = await authenticatedFetch(`/cash/shift/close`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload),
  });
  if (!res.ok) {
    const err = await extractData(res);
    throw new Error((err as any).message || "Failed to close shift");
  }
  return extractData(res);
}

export async function getCurrentShift(shopId: string): Promise<ShiftClosing | null> {
  const res = await authenticatedFetch(`/cash/shift/current?shopId=${shopId}`);
  if (!res.ok) throw new Error("Failed to fetch current shift");
  return extractData(res);
}

export async function getCashVariances(
  shopId: string,
  filters?: { status?: string; startDate?: string; endDate?: string }
): Promise<CashVariance[]> {
  const p = new URLSearchParams({ shopId });
  if (filters?.status) p.append("status", filters.status);
  const res = await authenticatedFetch(`/operations/daily-closing/variances?${p}`);
  if (!res.ok) throw new Error("Failed to fetch variances");
  return extractData(res);
}

export async function approveCashVariance(payload: {
  varianceId: string;
  shopId: string;
  notes?: string;
}): Promise<CashVariance> {
  const res = await authenticatedFetch(
    `/operations/daily-closing/variances/approve`,
    {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(payload),
    }
  );
  if (!res.ok) {
    const err = await extractData(res);
    throw new Error((err as any).message || "Failed to approve variance");
  }
  return extractData(res);
}

// ─── EXPENSES ─────────────────────────────────────────────────────────────────

export interface Expense {
  id: string;
  voucherId: string;
  shopId: string;
  amount: number;
  expenseCategory?: string;
  expenseCategoryId?: string;
  paymentMethod: string;
  narration?: string;
  date: string;
  status: string;
  createdAt: string;
}

export interface ExpenseCategoryBreakdown {
  category: string;
  total: number;
}

export async function createExpense(payload: {
  shopId: string;
  amount: number;
  category?: string;
  categoryId?: string;
  paymentMethod: string;
  note?: string;
  date?: string;
}): Promise<Expense> {
  const res = await authenticatedFetch(`/operations/expenses`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload),
  });
  if (!res.ok) {
    const err = await extractData(res);
    throw new Error((err as any).message || "Failed to create expense");
  }
  return extractData(res);
}

export async function getExpenseCategories(shopId: string): Promise<ExpenseCategory[]> {
  const res = await authenticatedFetch(`/operations/expenses/categories/list?shopId=${shopId}`);
  if (!res.ok) throw new Error("Failed to fetch expense categories");
  return extractData(res);
}

export async function seedExpenseCategories(): Promise<void> {
  await authenticatedFetch(`/operations/expenses/categories/seed`, { method: "POST" });
}

export async function getExpenses(
  shopId: string,
  filters?: {
    startDate?: string;
    endDate?: string;
    skip?: number;
    take?: number;
  }
): Promise<{ data: Expense[]; total: number }> {
  const p = new URLSearchParams({ shopId });
  if (filters?.startDate) p.append("startDate", filters.startDate);
  if (filters?.endDate) p.append("endDate", filters.endDate);
  if (filters?.skip !== undefined) p.append("skip", String(filters.skip));
  if (filters?.take !== undefined) p.append("take", String(filters.take));
  const res = await authenticatedFetch(`/operations/expenses?${p}`);
  if (!res.ok) throw new Error("Failed to fetch expenses");
  return extractData(res);
}

export async function getExpenseCategoryBreakdown(
  shopId: string,
  startDate: string,
  endDate: string
): Promise<ExpenseCategoryBreakdown[]> {
  const p = new URLSearchParams({ shopId, startDate, endDate });
  const res = await authenticatedFetch(`/operations/expenses/categories?${p}`);
  if (!res.ok) throw new Error("Failed to fetch category breakdown");
  return extractData(res);
}

// ─── STOCK VERIFICATION ───────────────────────────────────────────────────────

export interface StockVerificationSummary {
  totalItemsChecked: number;
  mismatchItems: number;
  totalLossValue: number;        // Paisa
  totalLossValueRupees: number;  // Rupees (ready to display)
}

export interface StockVerificationSession {
  id: string;
  shopId: string;
  sessionDate: string;
  status: StockVerificationStatus;
  notes?: string;
  createdBy: string;
  confirmedBy?: string;
  confirmedAt?: string;
  createdAt: string;
  _count?: { items: number };
  summary?: StockVerificationSummary;
}

export interface StockVerificationItem {
  id: string;
  shopProductId: string;
  systemQty: number;
  physicalQty: number;
  difference: number;
  reason?: AdjustmentReason;
  notes?: string;
  stockLedgerId?: string;
  shopProduct?: { name: string; category?: string; quantity: number; avgCost?: number };
}

export async function createVerificationSession(payload: {
  shopId: string;
  sessionDate: string;
  notes?: string;
}): Promise<StockVerificationSession> {
  const res = await authenticatedFetch(`/operations/stock-verification`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload),
  });
  if (!res.ok) {
    const err = await extractData(res);
    throw new Error((err as any).message || "Failed to create session");
  }
  return extractData(res);
}

export async function addVerificationItems(
  sessionId: string,
  items: Array<{
    shopProductId: string;
    physicalQty: number;
    reason?: AdjustmentReason;
    notes?: string;
  }>
): Promise<StockVerificationSession> {
  const res = await authenticatedFetch(
    `/operations/stock-verification/${sessionId}/items`,
    {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({ items }),
    }
  );
  if (!res.ok) {
    const err = await extractData(res);
    throw new Error((err as any).message || "Failed to add items");
  }
  return extractData(res);
}

export async function confirmVerificationSession(
  sessionId: string,
  shopId: string
): Promise<StockVerificationSession> {
  const res = await authenticatedFetch(
    `/operations/stock-verification/${sessionId}/confirm`,
    {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({ shopId }),
    }
  );
  if (!res.ok) {
    const err = await extractData(res);
    throw new Error((err as any).message || "Failed to confirm session");
  }
  return extractData(res);
}

export async function getVerificationSessions(
  shopId: string,
  filters?: { status?: string; startDate?: string; endDate?: string }
): Promise<StockVerificationSession[]> {
  const p = new URLSearchParams({ shopId });
  if (filters?.status) p.append("status", filters.status);
  if (filters?.startDate) p.append("startDate", filters.startDate);
  if (filters?.endDate) p.append("endDate", filters.endDate);
  const res = await authenticatedFetch(`/operations/stock-verification?${p}`);
  if (!res.ok) throw new Error("Failed to fetch sessions");
  return extractData(res);
}

export async function getVerificationSession(
  sessionId: string
): Promise<StockVerificationSession & { items: StockVerificationItem[] }> {
  const res = await authenticatedFetch(
    `/operations/stock-verification/${sessionId}`
  );
  if (!res.ok) throw new Error("Session not found");
  return extractData(res);
}

// ─── MONTHLY REPORT ───────────────────────────────────────────────────────────

export interface MonthlyProfit {
  totalSales: number;
  totalCogs: number;
  totalExpenses: number;
  totalRefunds: number;
  totalInvLoss: number;
  netProfit: number;
}

export interface MonthlyReport {
  period: { month: number; year: number; startDate: string; endDate: string };
  sales: { totalAmount: number; totalInvoices: number };
  purchases: { totalAmount: number; totalPurchases: number };
  expenses: { totalAmount: number; totalVouchers: number };
  salary: { totalAmount: number };
  refunds: { totalAmount: number };
  inventoryLoss: number;
  jobCards: { completed: number };
  profitSummary: {
    grossRevenue: number;
    totalCosts: number;
    netProfit: number;
    profitMarginPct: number;
  };
}

export async function getMonthlySummary(
  shopId: string,
  month: number,
  year: number
): Promise<MonthlyReport> {
  const p = new URLSearchParams({
    shopId,
    month: String(month),
    year: String(year),
  });
  const res = await authenticatedFetch(`/operations/monthly-report?${p}`);
  if (!res.ok) throw new Error("Failed to fetch monthly report");
  return extractData(res);
}

export async function getMonthlyProfit(
  shopId: string,
  month: number,
  year: number
): Promise<MonthlyProfit> {
  const p = new URLSearchParams({
    shopId,
    month: String(month),
    year: String(year),
  });
  const res = await authenticatedFetch(`/mobileshop/reports/monthly-profit?${p}`);
  if (!res.ok) throw new Error("Failed to fetch monthly profit");
  return extractData(res);
}

export async function getMonthlyTrend(
  shopId: string,
  months = 6
): Promise<MonthlyReport[]> {
  const p = new URLSearchParams({ shopId, months: String(months) });
  const res = await authenticatedFetch(`/operations/monthly-report/trend?${p}`);
  if (!res.ok) throw new Error("Failed to fetch trend");
  return extractData(res);
}

// ─── SHRINKAGE INTELLIGENCE ───────────────────────────────────────────────────

export interface ShrinkageByCategory {
  category: string;
  lostUnits: number;
  lostValue: number;
  affectedProducts: number;
}

export interface ShrinkageByStaff {
  staffId: string;
  staffName: string;
  staffEmail: string;
  sessions: number;
  lostUnits: number;
  lostValue: number;
}

export interface ShrinkageBySupplier {
  supplierId: string;
  supplierName: string;
  lostUnits: number;
  lostValue: number;
  affectedProducts: number;
}

export interface ShrinkageByReason {
  reason: string;
  count: number;
  lostUnits: number;
  lostValue: number;
}

export interface ShrinkageTopProduct {
  productId: string;
  productName: string;
  category: string;
  lossQty: number;
  lossValue: number;
}

export interface ShrinkageMonthlyTrend {
  month: string;
  lossValue: number;
  lossQty: number;
}

export interface ShrinkageIntelligence {
  period: { startDate: string; endDate: string };
  totalLostValue: number;
  totalLostUnits: number;
  topLossCategory: string | null;
  topLossStaff: string | null;
  topLossSupplier: string | null;
  byCategory: ShrinkageByCategory[];
  byStaff: ShrinkageByStaff[];
  bySupplier: ShrinkageBySupplier[];
  byReason: ShrinkageByReason[];
  topProducts: ShrinkageTopProduct[];
}

export async function getShrinkageIntelligence(
  shopId: string,
  startDate: string,
  endDate: string
): Promise<ShrinkageIntelligence> {
  const p = new URLSearchParams({ shopId, startDate, endDate });
  const res = await authenticatedFetch(`/operations/shrinkage/intelligence?${p}`);
  if (!res.ok) throw new Error("Failed to fetch shrinkage intelligence");
  return extractData(res);
}

export async function getShrinkageTopProducts(
  shopId: string,
  startDate: string,
  endDate: string,
  limit = 10
): Promise<ShrinkageTopProduct[]> {
  const p = new URLSearchParams({ shopId, startDate, endDate, limit: String(limit) });
  const res = await authenticatedFetch(`/operations/shrinkage/top-loss-products?${p}`);
  if (!res.ok) throw new Error("Failed to fetch top loss products");
  return extractData(res);
}

export async function getShrinkageMonthlyTrend(
  shopId: string,
  months = 12
): Promise<ShrinkageMonthlyTrend[]> {
  const p = new URLSearchParams({ shopId, months: String(months) });
  const res = await authenticatedFetch(`/operations/shrinkage/monthly-trend?${p}`);
  if (!res.ok) throw new Error("Failed to fetch shrinkage trend");
  return extractData(res);
}
// ─── EXPENSE INTELLIGENCE ────────────────────────────────────────────────────

export interface ExpenseIntelligenceOverview {
  totalExpense: number;
  averageDailyExpense: number;
  highestExpenseDay: {
    date: string;
    amount: number;
  };
}

export interface ExpenseIntelligenceCategory {
  category: string;
  amount: number;
}

export interface ExpenseIntelligenceTrend {
  month: string;
  amount: number;
}

export interface ExpenseIntelligencePayment {
  method: string;
  amount: number;
}

export interface ExpenseIntelligence {
  overview: ExpenseIntelligenceOverview;
  categoryBreakdown: ExpenseIntelligenceCategory[];
  monthlyTrend: ExpenseIntelligenceTrend[];
  paymentMethods: ExpenseIntelligencePayment[];
  insights: { insights: string[] };
}

export async function getExpenseIntelligence(
  shopId: string,
  startDate?: string,
  endDate?: string
): Promise<ExpenseIntelligence> {
  const p = new URLSearchParams({ shopId });
  if (startDate) p.append("startDate", startDate);
  if (endDate) p.append("endDate", endDate);
  
  const res = await authenticatedFetch(`/reports/expense-intelligence?${p}`);
  if (!res.ok) throw new Error("Failed to fetch expense intelligence");
  return extractData(res);
}

File: apps/mobibix-android/app/src/main/java/com/aiyal/mobibix/data/network/OperationsApi.kt
package com.aiyal.mobibix.data.network

import retrofit2.http.*

// ─── Expense Models ───────────────────────────────────────────────────────────

data class ExpenseCategory(
    val id: String,
    val name: String,
    val isDefault: Boolean = false
)

data class Expense(
    val id: String,
    val shopId: String,
    val categoryId: String? = null,
    val categoryName: String? = null,
    val amount: Double,
    val description: String,
    val paymentMode: String,
    val date: String,
    val staffName: String? = null,
    val createdAt: String? = null
)

data class ExpenseCategoryBreakdown(
    val categoryName: String,
    val total: Double,
    val count: Int,
    val percent: Double
)

data class CreateExpenseDto(
    val shopId: String,
    val categoryId: String? = null,
    val amount: Int, // integer Rupees — backend converts to Paisa
    val paymentMethod: String = "CASH",
    val note: String? = null,
    val date: String? = null
)

data class ExpenseListResponse(val data: List<Expense> = emptyList(), val total: Double = 0.0)
data class CategoryBreakdownResponse(val data: List<ExpenseCategoryBreakdown> = emptyList())

// ─── Daily Closing Models ─────────────────────────────────────────────────────

data class DailySummary(
    val date: String,
    val shopId: String,
    val status: String = "OPEN",
    val openingCash: Double = 0.0,
    val salesCash: Double = 0.0,
    val salesUpi: Double = 0.0,
    val salesCard: Double = 0.0,
    val salesBank: Double = 0.0,
    val cashWithdrawFromBank: Double = 0.0,
    val cashDepositToBank: Double = 0.0,
    val otherCashIn: Double = 0.0,
    val otherCashOut: Double = 0.0,
    val totalIn: Double = 0.0,
    val totalOut: Double = 0.0,
    val supplierPaymentsCash: Double = 0.0,
    val expenseCash: Double = 0.0,
    val expectedClosingCash: Double = 0.0,
    val reportedClosingCash: Double = 0.0,
    val cashDifference: Double = 0.0
)

data class DailyClosing(
    val id: String,
    val shopId: String,
    val date: String,
    val status: String,
    val openingCash: Double = 0.0,
    val salesCash: Double = 0.0,
    val salesUpi: Double = 0.0,
    val salesCard: Double = 0.0,
    val salesBank: Double = 0.0,
    val expectedClosingCash: Double = 0.0,
    val reportedClosingCash: Double = 0.0,
    val cashDifference: Double = 0.0,
    val varianceReason: String? = null,
    val varianceNote: String? = null,
    val closedAt: String? = null,
    val createdAt: String? = null
)

data class DailyClosingManualEntries(
    val salesCash: Double? = null,
    val salesUpi: Double? = null,
    val salesCard: Double? = null,
    val salesBank: Double? = null,
    val otherCashIn: Double? = null,
    val cashWithdrawFromBank: Double? = null,
    val expenseCash: Double? = null,
    val supplierPaymentsCash: Double? = null,
    val otherCashOut: Double? = null,
    val cashDepositToBank: Double? = null
)

data class SubmitDailyClosingDto(
    val shopId: String,
    val date: String,
    val mode: String = "SYSTEM", // SYSTEM or MANUAL
    val reportedClosingCash: Double,
    val manualEntries: DailyClosingManualEntries? = null,
    val varianceReason: String? = null,
    val varianceNote: String? = null
)

// ─── Stock Verification Models ────────────────────────────────────────────────

data class StockVerificationItem(
    val shopProductId: String,
    val productName: String,
    val systemQty: Int,
    val countedQty: Int? = null,
    val variance: Int = 0
)

data class StockVerificationSession(
    val id: String,
    val shopId: String,
    val status: String,
    val createdAt: String,
    val confirmedAt: String? = null,
    val items: List<StockVerificationItem>? = null
)

data class StartStockVerificationDto(val shopId: String)
data class UpdateStockCountDto(val shopProductId: String, val countedQty: Int)
data class ConfirmStockVerificationDto(val sessionId: String)

// ─── Monthly Report Models ────────────────────────────────────────────────────

data class MonthlyProfit(
    val month: String,
    val revenue: Double,
    val expenses: Double,
    val profit: Double
)

data class MonthlyReport(
    val month: String,
    val shopId: String,
    val totalRevenue: Double,
    val totalExpenses: Double,
    val netProfit: Double,
    val salesCount: Int,
    val repairCount: Int
)

// ─── Shrinkage Models ─────────────────────────────────────────────────────────

data class ShrinkageByCategory(val category: String, val lossQty: Int, val lossValue: Double, val percent: Double)
data class ShrinkageByReason(val reason: String, val count: Int, val lossQty: Int, val lossValue: Double)
data class ShrinkageTopProduct(val productId: String, val productName: String, val lossQty: Int, val lossValue: Double)
data class ShrinkageMonthlyTrend(val month: String, val lossQty: Int, val lossValue: Double)

data class ShrinkageIntelligence(
    val totalLossValue: Double = 0.0,
    val totalLossQty: Int = 0,
    val topReason: String? = null,
    val byCategory: List<ShrinkageByCategory> = emptyList(),
    val byReason: List<ShrinkageByReason> = emptyList(),
    val topProducts: List<ShrinkageTopProduct> = emptyList(),
    val monthlyTrend: List<ShrinkageMonthlyTrend> = emptyList()
)

// ─── Expense Intelligence Models ──────────────────────────────────────────────

data class ExpenseIntelligenceCategory(val name: String, val total: Double, val count: Int, val percent: Double)
data class ExpenseIntelligenceTrend(val month: String, val total: Double)

data class ExpenseIntelligence(
    val totalExpenses: Double = 0.0,
    val avgMonthly: Double = 0.0,
    val topCategory: String? = null,
    val byCategory: List<ExpenseIntelligenceCategory> = emptyList(),
    val trend: List<ExpenseIntelligenceTrend> = emptyList()
)

// ─── API ─────────────────────────────────────────────────────────────────────

interface OperationsApi {

    // Expenses
    @POST("api/operations/expenses")
    suspend fun createExpense(@Body dto: CreateExpenseDto): Expense

    @GET("api/operations/expenses")
    suspend fun listExpenses(
        @Query("shopId") shopId: String,
        @Query("startDate") startDate: String? = null,
        @Query("endDate") endDate: String? = null,
        @Query("categoryId") categoryId: String? = null
    ): ExpenseListResponse

    @GET("api/operations/expenses/categories/list")
    suspend fun listExpenseCategories(@Query("shopId") shopId: String): List<ExpenseCategory>

    @GET("api/operations/expenses/categories")
    suspend fun getExpenseCategoryBreakdown(
        @Query("shopId") shopId: String,
        @Query("startDate") startDate: String? = null,
        @Query("endDate") endDate: String? = null
    ): CategoryBreakdownResponse

    // Daily Closing
    @GET("api/cash/daily-history")
    suspend fun getDailyHistory(@Query("shopId") shopId: String): List<DailyClosing>

    @POST("api/cash/daily-close")
    suspend fun submitDailyClosing(@Body dto: SubmitDailyClosingDto): DailyClosing

    @GET("api/mobileshop/reports/daily-summary")
    suspend fun getDailySummary(@Query("shopId") shopId: String, @Query("date") date: String): DailySummary

    // Stock Verification
    @POST("api/operations/stock-verification")
    suspend fun startStockVerification(@Body dto: StartStockVerificationDto): StockVerificationSession

    @GET("api/operations/stock-verification")
    suspend fun listStockVerifications(@Query("shopId") shopId: String): List<StockVerificationSession>

    @PATCH("api/operations/stock-verification/{id}/items")
    suspend fun updateStockCount(
        @Path("id") sessionId: String,
        @Body dto: UpdateStockCountDto
    ): StockVerificationItem

    @POST("api/operations/stock-verification/{id}/confirm")
    suspend fun confirmStockVerification(@Path("id") sessionId: String): StockVerificationSession

    // Monthly Report
    @GET("api/operations/monthly-report")
    suspend fun getMonthlyReport(
        @Query("shopId") shopId: String,
        @Query("month") month: String? = null,
        @Query("year") year: Int? = null
    ): MonthlyReport

    @GET("api/operations/monthly-report/trend")
    suspend fun getMonthlyTrend(
        @Query("shopId") shopId: String,
        @Query("months") months: Int = 6
    ): List<MonthlyProfit>

    // Shrinkage Intelligence
    @GET("api/operations/shrinkage/intelligence")
    suspend fun getShrinkageIntelligence(
        @Query("shopId") shopId: String,
        @Query("startDate") startDate: String? = null,
        @Query("endDate") endDate: String? = null
    ): ShrinkageIntelligence

    @GET("api/operations/shrinkage/top-loss-products")
    suspend fun getShrinkageTopProducts(
        @Query("shopId") shopId: String,
        @Query("startDate") startDate: String? = null,
        @Query("endDate") endDate: String? = null
    ): List<ShrinkageTopProduct>

    @GET("api/operations/shrinkage/monthly-trend")
    suspend fun getShrinkageMonthlyTrend(
        @Query("shopId") shopId: String
    ): List<ShrinkageMonthlyTrend>

    // Expense Intelligence
    @GET("api/reports/expense-intelligence")
    suspend fun getExpenseIntelligence(
        @Query("shopId") shopId: String,
        @Query("startDate") startDate: String? = null,
        @Query("endDate") endDate: String? = null
    ): ExpenseIntelligence
}
